crates/arkworks/src/lib.rs

```diff
@@ -41,29 +40,34 @@ pub enum Result {
     Err,
 }
 
-impl From <std::result::Result<Vec<u8>, ()>> for Result {
+impl From<std::result::Result<Vec<u8>, ()>> for Result {
     fn from(value: std::result::Result<Vec<u8>, ()>) -> Self {
         match value {
             Ok(mut vec) => {
-                let r = BytesVec{
+                let r = BytesVec {
                     data: vec.as_mut_ptr(),
                     size: vec.len() as _,
                 };
                 std::mem::forget(vec);
                 Result::Ok(r)
-            },
-            Err(_) => Result::Err
+            }
+            Err(_) => Result::Err,
         }
     }
 }
 
 impl BytesVec {
     unsafe fn as_slice(&self) -> Vec<u8> {
-        std::slice::from_raw_parts_mut(self.data, self.size as usize).to_vec()
+        // Calling code may pass a null pointer for empty slices, but Rust expects an aligned pointer
```

Comment:
Deduplicate this comment and function.
Move into separate shared crate like `make_slice = { file = "../make_slice" }` (Cargo.toml)